openstudiocore/src/utilities/bcl/RemoteBCL.cpp

```diff
@@ -255,84 +219,86 @@ namespace openstudio{
   int RemoteBCL::checkForComponentUpdates()
   {
     m_componentsWithUpdates.clear();
-/*
+
     for (const BCLComponent& component : LocalBCL::instance().components()) {
-      // can't start another search until the last one is done
-      if (!m_mutex.try_lock()){
-        return 0;
+      // can't start another request until last one is done
+      if (m_httpResponse && !m_httpResponse->is_done()) {
+        return false;
       }
 
-      // disconnect all signals from m_networkManager to this
-      //disconnect(m_networkManager, nullptr, this, nullptr); // returns a bool, but was not checking, so removed
-
-      const_cast<RemoteBCL*>(this)->m_lastSearch.clear();
-
-      //QString url = toQString(remoteUrl() + "/api/search/?fq[]=ss_uuid:%1&api_version=%2").arg(
-      //  toQString(component.uid()),
-      //  toQString(m_apiVersion)
-      //);
-      //LOG(Warn, toString(url));
-
-      // when the reply is finished call onSearchResponseComplete
-      //connect(m_networkManager, &QNetworkAccessManager::finished, this, &RemoteBCL::onSearchResponseComplete);
-
-//#ifndef QT_NO_OPENSSL
-//      connect(m_networkManager, &QNetworkAccessManager::sslErrors, this, &RemoteBCL::catchSslErrors);
-//#endif
-*/
-      //QNetworkRequest request = QNetworkRequest(QUrl(url));
-      //request.setRawHeader("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8");
-/* 
-      //request.setRawHeader("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.17 (KHTML, like Gecko) Chrome/24.0.1312.56 Safari/537.17");
-      //m_networkManager->get(request);
-      //std::vector<BCLSearchResult> result = waitForSearch();
-      //if (result.size() > 0 && result[0].versionId() != component.versionId()){
-       // m_componentsWithUpdates.push_back(result[0]);
-      //}
+      m_lastSearch.clear();
+
+      web::http::client::http_client client(toWString(remoteUrl()));
```

Comment:
@axelstudios does this client use wstring on all platforms?

Reply:
If so, does it assume UTF-16 encoding?  Which byte order?  We need to make sure toWString does the right conversion.